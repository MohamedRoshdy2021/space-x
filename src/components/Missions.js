function Missions() {
  return (
    <div>
      <h1>Missions Page</h1>
    </div>
  );
}
export default Missions;
