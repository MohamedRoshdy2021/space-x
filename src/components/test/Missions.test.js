import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Missions from '../Mission/mission'

const mockStore = configureStore([thunk]);

test('renders Missions correctly', () => {
  const store = mockStore({
    missions: {
      isLoading: false,
      missions: [
        { id: 1, name: 'Mission 1', description: 'Description 1' },
        { id: 2, name: 'Mission 2', description: 'Description 2' },
      ],
      joinedMissions: [1],
    },
  });

  const { asFragment } = render(
    <Provider store={store}>
      <Missions />
    </Provider>,
  );

  expect(asFragment()).toMatchSnapshot();
});