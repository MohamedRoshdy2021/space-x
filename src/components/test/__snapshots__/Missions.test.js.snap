// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Missions correctly 1`] = `
<DocumentFragment>
  <table
    class="missions"
  >
    <thead>
      <tr>
        <th
          class="mission"
        >
          Mission
        </th>
        <th
          class="description"
        >
          Description
        </th>
        <th
          class="status"
        >
          Status
        </th>
        <th
          class="button"
        >
          Â 
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="rows"
      >
        <td
          class="td-mission"
        >
          Mission 1
        </td>
        <td
          class="description td-description"
        >
          Description 1
        </td>
        <td
          class="status"
        >
          <span
            class="mission-status-leave"
          >
            Active Member
          </span>
        </td>
        <td
          class="button"
        >
          <button
            class="td-leave-mission"
            type="button"
          >
            Leave Mission
          </button>
        </td>
      </tr>
      <tr
        class="join-rows"
      >
        <td
          class="td-mission"
        >
          Mission 2
        </td>
        <td
          class="description td-description"
        >
          Description 2
        </td>
        <td
          class="status"
        >
          <span
            class="mission-status-join"
          >
            NOT A MEMBER
          </span>
        </td>
        <td
          class="button"
        >
          <button
            class="td-join-mission"
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
